google/resource_dataflow_job.go

```diff
@@ -353,3 +361,33 @@ func resourceDataflowJobUpdateJob(config *Config, project string, region string,
 	}
 	return config.clientDataflow.Projects.Locations.Jobs.Update(project, region, id, job).Do()
 }
+
+func resourceDataflowJobSubnetworkDiffSuppress(_, old, new string, _ *schema.ResourceData) bool {
+	if match := validSubnetFormatRegex.FindString(new); match != "" {
+		oldMatch := validSubnetFormatRegex.FindString(old)
+		return oldMatch != "" && oldMatch == match
+	}
+	return false
+}
+
+func getSubnetwork(d *schema.ResourceData) (string, error) {
+	subnetwork := d.Get("subnetwork").(string)
+	if subnetwork == "" {
+		return "", nil
+	}
+	return toValidSubnetworkFormat(subnetwork)
+}
+
+// toValidSubnetworkFormat converts the given string to a valid subnetwork
+// string recognizable by the Dataflow API (i.e. in the format of
+// "regions/REGION/subnetworks/SUBNETWORK"). Returns an error if not possible.
+// This function exists to allow subnetworks to be specified as one of:
+// (1) self_links: https://www.googleapis.com/compute/v1/projects/PROJECT/regions/REGION/subnetworks/SUBNETWORK
+// (2) partial URIs: projects/PROJECT/regions/REGION/subnetworks/SUBNETWORK
+// (3) Dataflow subnetwork format: regions/REGION/subnetworks/SUBNETWORK
+func toValidSubnetworkFormat(subnetwork string) (string, error) {
+	if match := validSubnetFormatRegex.FindString(subnetwork); match != "" {
```

Comment:
Hi @emilymye, `ParseSubnetworkFieldValue()` returns a `RegionalFieldValue` which can be converted to a relative link using its `RelativeLink()` method (i.e. `projects/{project}/regions/{region}/subnetworks/{subnetwork}`) .

However, the underlying API only accepts either the full URL or `regions/{region}/subnetworks/{subnetwork}` ([source](https://cloud.google.com/dataflow/docs/guides/specifying-networks#subnetwork_parameter)).

Is there a good way to convert a subnetwork relative link to a full URL?

Reply:
I could also just trim the `projects/{project}` prefix (and check that the subnetwork's project is the same as the job's project so that the user isn't misled that they can reference a subnetwork from another project), but was wondering if we can pass a full URL instead.

Reply:
I'm mostly trying to reuse parsing utils, which makes it easier to present the same interface to Terraform users, i.e. they can provide resource URLs in config as self-link, partial resource names, or resource name, rather than forcing them to know per-API what format they need to provide.

If we want to hardcode the format into Terraform, that's fine, but I'd rather it be just a format string that uses values from the parsed RegionalFieldValue to create the API-expected value, e.g. `fmt.Sprintf("regions/{region}/subnetworks/{subnetwork}", fv.region, fv.name)`. 

Is there a reason we have to pass full URL?

Reply:
Makes sense to me.

I wanted to pass the full URL to the underlying API since I did not want to mislead the user into thinking that they can reference a subnetwork from another project when we we're actually dropping the project in reality.

Also, according to [this](https://cloud.google.com/dataflow/docs/guides/specifying-networks#subnetwork_parameter), you need to specify the full URL if the subnetwork is in a Shared VPC network.